segmentation.py

```diff
@@ -197,6 +197,7 @@ def find_dividers(matrix: List[Path]) -> Tuple[pd.DataFrame, Set[int]]:
             'to': path_dividers[:, 1],    # aka downstream
             'path_index': i
         })
+        df = utils.sort_and_drop_duplicates(df) # early deduplication saves lots of runtime memory
```

Comment:
Then you should also change `n_remaining_links` calculation, it's supposed to output total amount of links including duplicates (just update it before de-duping here).

Reply:
Done